jaxrs/src/main/java/com/cerner/beadledom/jaxrs/provider/ForwardedHeaderFilter.java

```diff
@@ -0,0 +1,72 @@
+package com.cerner.beadledom.jaxrs.provider;
+
+import java.io.IOException;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import javax.annotation.Priority;
+import javax.ws.rs.Priorities;
+import javax.ws.rs.container.ContainerRequestContext;
+import javax.ws.rs.container.ContainerRequestFilter;
+import javax.ws.rs.container.PreMatching;
+import javax.ws.rs.ext.Provider;
+
+/**
+ * The ForwardedHeaderFilter reads the request context and determines if the client request was made
+ * from a secure protocol (HTTPS). If the request originated from secure context, it updates the
+ * request to reflect the original protocol.
+ *
+ * <p>To determine if the client request made in a secure context was forwarded by a load balancer
+ * or proxy, the "Forwarded" and "X-Forwarded-Proto" values are extracted and checked.</p>
```

Comment:
It seems like it is deemed a secure context if one of the headers is present in the request and not both of them. Should we explicitly say that here in the description?

Reply:
I added a little clearer description: https://github.com/cerner/beadledom/pull/149/commits/8930d43b5dab6c50fc6b30305109bea73744d62a